src/main/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminService.java

```diff
@@ -2,81 +2,88 @@
 
 import be.jidoka.jdk.keycloak.admin.domain.AddClientRoleToUser;
 import be.jidoka.jdk.keycloak.admin.domain.CreateUser;
+import be.jidoka.jdk.keycloak.admin.domain.GetUserRequest;
+import be.jidoka.jdk.keycloak.admin.domain.GetUsersRequest;
 import be.jidoka.jdk.keycloak.admin.domain.RemoveClientRoleFromUser;
+import be.jidoka.jdk.keycloak.admin.domain.SearchUserRequest;
 import be.jidoka.jdk.keycloak.admin.domain.User;
+import org.apache.commons.lang3.StringUtils;
 import org.keycloak.admin.client.resource.ClientsResource;
 import org.keycloak.admin.client.resource.UserResource;
 import org.keycloak.admin.client.resource.UsersResource;
 import org.keycloak.representations.idm.RoleRepresentation;
 import org.keycloak.representations.idm.UserRepresentation;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
 
 import javax.ws.rs.core.Response;
+import java.util.HashMap;
 import java.util.List;
-import java.util.Set;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 import static be.jidoka.jdk.keycloak.admin.domain.User.PICTURE_URL_ATTRIBUTE;
 import static java.util.Collections.singletonList;
-import static java.util.Collections.singletonMap;
+import static java.util.stream.Collectors.toList;
+import static org.keycloak.admin.client.CreatedResponseUtil.getCreatedId;
 
 public class KeycloakUserAdminService {
 
-	public static final String USER_ID_FROM_PATH_REGEX = ".*/([^/]+)$";
-
 	private final UsersResource usersResource;
 	private final ClientsResource clientsResource;
+	private final UserFactory userFactory;
 
 	public KeycloakUserAdminService(UsersResource usersResource, ClientsResource clientsResource) {
 		this.usersResource = usersResource;
 		this.clientsResource = clientsResource;
+		this.userFactory = new UserFactory();
 	}
 
-	public Set<User> getUsers() {
-		return usersResource.list().stream()
-				.map(userRepresentation -> new User(
-						userRepresentation.getId(),
-						userRepresentation.getUsername(),
-						userRepresentation.getFirstName(),
-						userRepresentation.getLastName(),
-						userRepresentation.getEmail(),
-						userRepresentation.getAttributes()
-				))
-				.collect(Collectors.toSet());
+	public Page<User> getUsers(GetUsersRequest request) {
```

Comment:
Missing test for this method, specifically the paging.

Reply:
Done.